test/test.oauth2.ts

```diff
@@ -1338,15 +1378,95 @@ describe('oauth2', () => {
       assert.deepStrictEqual(info.scopes, tokenInfo.scope.split(' '));
     });
 
-    it('should throw if tries to refresh but no refresh token is available', async () => {
+    it('should call refreshHandler in getRequestHeaders() on token expiration and refreshHandler is available', async () => {
+      const expectedRefreshedAccessToken = {
+        access_token: 'access_token',
+        expiry_date: new Date().getTime() + 3600 * 1000,
+      };
+      client.refreshHandler = async () => {
+        return expectedRefreshedAccessToken;
+      };
       client.setCredentials({
         access_token: 'initial-access-token',
         expiry_date: new Date().getTime() - 1000,
       });
+      const expectedMetadata = {
+        Authorization: 'Bearer access_token',
+      };
+
+      const requestMetaData = await client.getRequestHeaders(
+        'http://example.com'
+      );
+
+      assert.deepStrictEqual(requestMetaData, expectedMetadata);
+    });
+
+    it('should return authorization header on current access token if it has not expired', async () => {
+      client.setCredentials({
+        access_token: 'initial-access-token',
+        expiry_date: new Date().getTime() + 3600 * 1000,
+      });
+      const expectedMetadata = {
+        Authorization: 'Bearer initial-access-token',
+      };
+
+      const requestMetaData = await client.getRequestHeaders(
+        'http://example.com'
+      );
+
+      assert.deepStrictEqual(requestMetaData, expectedMetadata);
+    });
+
+    it('should throw if tries to refresh but no refreshHandler callback or refresh token is available', async () => {
+      client.setCredentials({
+        access_token: 'initial-access-token',
+        expiry_date: new Date().getTime() - 1000,
+      });
+
       await assert.rejects(
         client.getRequestHeaders('http://example.com'),
         /No refresh token is set./
       );
     });
+
+    it('should call refreshHandler in getAccessToken() on expiration and no refresh token available', async () => {
+      const expectedRefreshedAccessToken = {
+        access_token: 'access_token',
+        expiry_date: new Date().getTime() + 3600 * 1000,
+      };
+      client.refreshHandler = async () => {
+        return expectedRefreshedAccessToken;
+      };
+      client.setCredentials({
```

Comment:
All your tests for `refreshHandler` rely on cached credentials being available. Please add tests when no credentials are available (don't use `setCredentials`)

Add tests for:
- `getAccessToken()`
- `request()`
- `getRequestHeaders()`